modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -1665,6 +1667,15 @@ private File pollNextFile(long curIdx) throws StorageException, IgniteInterrupte
         return scan(walArchiveDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER));
     }
 
+    /**
+     * Files from {@link #walWorkDir}.
+     *
+     * @return Raw WAL segments from working directory.
+     */
+    public FileDescriptor[] walWorkFiles() {
+        return scan(walWorkDir.listFiles(WAL_SEGMENT_FILE_FILTER));
```

Comment:
Please do the same as in the **FileWriteAheadLogManager#walArchiveFiles**:
`return scan(walWorkDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER));`

It is also recommended to add new methods to the end of the class.

Reply:
In general, you can not add a new method **FileWriteAheadLogManager#walWorkFiles**, but reuse the **FileWriteAheadLogManager#loadFileDescriptors**, make a method in the test that we will remake the **IgniteCheckedException** into **IgniteException**.